src/main/java/vivatech/util/TierHelper.java

```diff
@@ -0,0 +1,45 @@
+package vivatech.util;
+
+import java.lang.reflect.InvocationTargetException;
+
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import vivatech.Vivatech;
+import vivatech.block.AbstractTieredMachineBlock;
+
+public class TierHelper {
+
+	public static Identifier getTieredID(Identifier id, MachineTier tier) {
+		return new Identifier(id.getNamespace(), tier.getAffix() + "_" + id.getPath());
+	}
+	
+	public static AbstractTieredMachineBlock[] fillTieredBlockArray(Class<? extends AbstractTieredMachineBlock> clazz, AbstractTieredMachineBlock[] array) {
+		for(MachineTier tier : MachineTier.values()) {
+			try {
+				array[tier.ordinal()] = clazz.getConstructor(MachineTier.class).newInstance(tier);
```

Comment:
....or, you could just do something like
```java
public static AbstractTieredMachineBlock[] fillTieredBlockArray(Function<MachineTier, ? extends AbstractTieredMachineBlock> function, MachineTier... tiers) {
    AbstractTieredMachineBlock[] result = new AbstractTieredMachineBlock[tiers.length];
    for(int i=0; i<tiers.size; i++) result[i] = function.apply(tiers[i]);
    return result;
}

(...)

ELECTRIC_FURNACE = fillTieredBlockArray(ElectricFurnaceBlock::new, MachineTier.values());
```
and not have all the reflection and potential hard crashes that go with it.

Reply:
Yeah, that's a much better way of doing that. Thanks